### PR TITLE
Added authentication API and server

### PR DESCRIPTION
**Pull request by coconut750750**
- Integrated the authentication API into the project api
- Added the authentication server into the top level directory

### CODE REVIEW

File: api/routes/index.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 const express = require('express');
 const router = express.Router();
+var validate = require('express-jsonschema').validate;
```
Comment (yousefa00) [inline api/routes/index.js]: It may be better to have these as const since we want to avoid var

File: api/routes/index.js [hunk 2]
```diff
@@ -13,4 +14,51 @@ router.get('/', function(req, res, next) {
   res.send('Kids Save Ocean');
 });
 
+var ModelSchema = {
```
Comment (yousefa00) [inline api/routes/index.js]: This should also be `const` or at least use `let` since we want to avoid `var`

File: auth-server/config.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        - supervisor
+        - guest
+    admin: 
+        - supervisor
+        - guest
+    supervisor: 
+        - guest
+    guest: 
+security_questions:
+    - Do you love this shit?
```
Comment (yousefa00) [inline auth-server/config.yml]: LOL maybe leave these out

Comment (coconut750750) [reply auth-server/config.yml]: Oops yeah

**Comment by yousefa00** [comment]
Looks good! There are some minor linting issues that should be fixed before merging and you should uninstall jade to fix the audit issue (I have a fix in the create_model_endpoint branch which should be merged into master soon so you can get it there). Let me know if you'd like me to just commit the fix here instead.

**Pull request merged**